src/aosm/azext_aosm/cli_handlers/onboarding_base_handler.py

```diff
@@ -339,3 +334,12 @@ def _build_deploy_params_schema(schema_properties):
         }
         schema_contents["properties"] = schema_properties
         return schema_contents
+
+    def _validate_input_file(self, input_file: Path):
+        """Validate the input file."""
+        if not input_file.suffix == ".jsonc":

```

Comment:
I'm not really convinced checking the file extension is the right way to validate this. It's also very Windows centric - Linux doesn't care about file extensions.

Better would be to try to load the file, and let the JSON library exception be bubbled up.